develop/parachain-devs/runtime-development/FRAME/add-pallet-to-the-runtime.md

```diff
@@ -0,0 +1,143 @@
+
+At its core, the `Config` trait typically looks like this:
+
+```rust
+--8<-- 'code/develop/parachain-devs/runtime-development/FRAME/add-pallet-to-the-runtime/pallet-basic-config-trait.rs'
+```
+
+This basic structure shows that every pallet must define certain types, such as `RuntimeEvent` and `WeightInfo`, to be functional within the runtime. The actual implementation can vary depending on the pallet’s specific needs.
+
+??? "Example - Utility Pallet"
+      For instance, in the [`utility pallet`](https://github.com/paritytech/polkadot-sdk/tree/master/substrate/frame/utility){target=\_blank}, the Config trait is implemented with the following types:
+
+      ```rust
+      --8<-- 'code/develop/parachain-devs/runtime-development/FRAME/add-pallet-to-the-runtime/utility-pallet-config-trait.rs'
+      ```
+
+     This example shows how the `Config` trait defines types like `RuntimeEvent`, `RuntimeCall`, `PalletsOrigin`, and `WeightInfo`, which the pallet will use when interacting with the runtime.
+
+## Parameters Configuration for Pallets
+
+Traits in Rust define shared behavior, and within the Polkadot SDK, they allow runtimes to integrate and utilize a pallet's functionality by implementing its associated configuration trait and parameters. Some of these parameters may require constant values, which can be defined using the [`parameter_types!`](https://paritytech.github.io/polkadot-sdk/master/frame_support/macro.parameter_types.html){target=\_blank} macro. This macro simplifies development by expanding the constants into the appropriate struct types with functions that the runtime can use to access their types and values in a consistent manner.
+
+For example, the following code snippet shows how the Solochain template configures certain parameters through the `parameter_types!` macro in the `runtime/lib.rs` file:
+
+```rust
+--8<-- 'code/develop/parachain-devs/runtime-development/FRAME/add-pallet-to-the-runtime/parameter-types-example.rs'
+```
+
+## Pallet Config in the Runtime
+
+To integrate a new pallet into the runtime, you need to implement its `Config` trait in the `runtime/lib.rs` file. This is done by specifying the necessary types and parameters in Rust, as shown below:
+
+```rust
+--8<-- 'code/develop/parachain-devs/runtime-development/FRAME/add-pallet-to-the-runtime/impl-pallet-example-in-runtime.rs'
+```
+
+Finally, to compose the runtime, update the list of pallets in the same file by modifying the [`#[frame_support::runtime]`](https://paritytech.github.io/polkadot-sdk/master/frame_support/attr.runtime.html){target=_blank} section. This Rust macro constructs the runtime with a specified name and the pallets you want to include. Use the following format when adding your pallet:
+
+```rust
+--8<-- 'code/develop/parachain-devs/runtime-development/FRAME/add-pallet-to-the-runtime/frame-support-runtime-macro.rs'
+```
+
+!!! note
```

Comment:
I would like a final section that concludes the guide explaining that now the pallet has been added, and the runtime can be compiled and used